### PR TITLE
Add LoRa-E5 / stm32wlex

### PR DESCRIPTION
**Pull request by kenbell**
Credit goes to @ofauchon for this code - this is borrowed from their branch.

I've cut it down to just the logic i'm able to verify locally (so no SPI, no LoRa) - and I've added a couple more timers.

### CODE REVIEW

**Comment by ofauchon** [comment]
Thanks @kenbell  for solving the issues with the timers. 

I made some quick tests on a Lora E5 device (timers, uart, GPIO)
Everything works fine.

**Comment by kenbell** [comment]
Awesome

@deadprogram - I think this one is good to go, resolves #1939

File: src/machine/machine_stm32wle5.go [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+	config.RX.ConfigureAltFunc(PinConfig{Mode: PinModeUARTRX}, uart.RxAltFuncSelector)
+}
+
+// UART baudrate calc based on the bus and clockspeed
+// NOTE: keep this in sync with the runtime/runtime_stm32wle5.go clock init code
+func (uart *UART) getBaudRateDivisor(baudRate uint32) uint32 {
+	var br uint32
+	uartClock := CPUFrequency() // No Prescaler configuration
+	br = uint32((uartClock + baudRate/2) / baudRate)
+	//br = uint32(uartClock / baudRate)
```
Comment (deadprogram) [inline src/machine/machine_stm32wle5.go]: Remove comment, please.

**Comment by deadprogram** [comment]
I made a few comments and a question. Otherwise looks good to me.

**Comment by kenbell** [comment]
@deadprogram - thanks for the review.  I only saw one comment, but i've been through and tidied up a couple of the code comments and removed the 'TODO' block with a better explanation comment that the clock will run at reset default (4 Mhz) for now.

@ofauchon should have some changes to finish the implementation

File: src/machine/machine_stm32wle5.go (hunk 1 above)
Comment (kenbell) [reply src/machine/machine_stm32wle5.go]: done

**Comment by deadprogram** [comment]
Thank you @ofauchon for starting and @kenbell for finishing this contribution! Now merging.

**Pull request merged**